src/media/local-track.ts

```diff
@@ -39,11 +24,8 @@ export type TrackPublishEvent = {
   trackState: TrackState;
 };
 
-export interface TrackEvents extends EventMap {
-  [LocalTrackEvents.Ended]: (event: TrackEndEvent) => void;
-  [LocalTrackEvents.Muted]: (event: TrackMuteEvent) => void;
-  [LocalTrackEvents.PublishedStateUpdate]: (event: TrackPublishEvent) => void;
-  [LocalTrackEvents.UnderlyingTrackChange]: () => void;
+export interface LocalTrackEvents extends EventMap {
+  [Events.PublishedStateUpdate]: (event: TrackPublishEvent) => void;
```

Comment:
Don't we still want the other events?

Reply:
Ah, I see they moved up to `Track`

Reply:
I think `UnderlyingTrackChange` will only apply for local tracks?

Reply:
the underlying track update will be on the track class , which is the one saving the local and track with effect

Reply:
> the underlying track update will be on the track class , which is the one saving the local and track with effect

sorry, don't think I understand.  `UnderlyingTrackChange` only applies to local tracks, and therefore should be defined at the `LocalTrack` level, not the `Track` level, right?

Reply:
thats true, The main problem is the track change action can happen from localCameraTrack and local-track for effect

which indirectly calls the track to change the underlying track and switch it , moving it to local-track, i might have to move all the effects back to local-track which will cause remoteTrack to not have effect 

i dont think its going to make much different if are making sure the change event is getting triggered from one place

Reply:
I don't think I understand.  Nothing that isn't a `LocalTrack` should be firing `UnderlyingTrackChange`, right? So what's the issue with moving the `UnderlyingTrackChange` event to `LocalTrack`?  Something with effects? Do we have support for effects on remote tracks?